<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="BasicR">
<title>S4 classes • BasicR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="S4 classes">
<meta property="og:description" content="BasicR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">BasicR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/R_data_types.html">R Data types</a>
    <a class="dropdown-item" href="../articles/S4_classes.html">S4 classes</a>
    <a class="dropdown-item" href="../articles/S4_classes_exercises.html">S4 classes, exercises</a>
    <a class="dropdown-item" href="../articles/Statistics.html">Statistics</a>
    <a class="dropdown-item" href="../articles/Statstics_solutions.html">Statistics_solutions</a>
    <a class="dropdown-item" href="../articles/assignment.html">Assignment</a>
    <a class="dropdown-item" href="../articles/basic_functions.html">Basic functions</a>
    <a class="dropdown-item" href="../articles/control_structures.html">Control structures</a>
    <a class="dropdown-item" href="../articles/data_cleaning.html">Data cleaning with Tidyverse</a>
    <a class="dropdown-item" href="../articles/data_cleaning_solutions.html">Data cleaning, solutions</a>
    <a class="dropdown-item" href="../articles/data_types_exercises.html">Data types, exercises</a>
    <a class="dropdown-item" href="../articles/data_types_exercises_solutions.html">Data types, solutions</a>
    <a class="dropdown-item" href="../articles/enrichment.html">Enrichment analysis</a>
    <a class="dropdown-item" href="../articles/exercises_control_exercises.html">Control structures, exercises</a>
    <a class="dropdown-item" href="../articles/exercises_control_solutions.html">Control structures, solutions</a>
    <a class="dropdown-item" href="../articles/functions_exercises.html">Functions, exercises</a>
    <a class="dropdown-item" href="../articles/functions_exercises_solutions.html">Functions, solutions</a>
    <a class="dropdown-item" href="../articles/ggplot2.html">GGplot2</a>
    <a class="dropdown-item" href="../articles/ggplot_solutions.html">GGplot solutions</a>
    <a class="dropdown-item" href="../articles/intro.html">Course introduction</a>
    <a class="dropdown-item" href="../articles/methylation_analysisRmd.html">DNA methylation</a>
    <a class="dropdown-item" href="../articles/r_functions.html">Functions, useful packages</a>
    <a class="dropdown-item" href="../articles/references.html">References</a>
    <a class="dropdown-item" href="../articles/reproducible.html">Reproducible programming</a>
    <a class="dropdown-item" href="../articles/reproducible_solutions.html">Reproducible programming, exercises</a>
    <a class="dropdown-item" href="../articles/resources.html">Resources</a>
    <a class="dropdown-item" href="../articles/set_up_git.html">Set up git</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/tkik/BasicR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="S4_classes_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>S4 classes</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/tkik/BasicR/blob/HEAD/vignettes/S4_classes.Rmd" class="external-link"><code>vignettes/S4_classes.Rmd</code></a></small>
      <div class="d-none name"><code>S4_classes.Rmd</code></div>
    </div>

    
    
<p>S4 classes are the perfect examples of the OO programming: - stricter structure - they have representations: a list of slots or arguments by name and class - the slots have to have the correct type - methods - they also inherit the methods from the classes of the arguments, but inheritence is between S4 classes as well. - has to be created with the new constructor - they have accessor methods as well.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">setClass</span><span class="op">(</span><span class="st">"Person"</span>, <span class="fu">representation</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"character"</span>, age <span class="op">=</span> <span class="st">"numeric"</span><span class="op">)</span><span class="op">)</span>
<span class="kw">setClass</span><span class="op">(</span><span class="st">"Employee"</span>, <span class="fu">representation</span><span class="op">(</span>boss <span class="op">=</span> <span class="st">"Person"</span><span class="op">)</span>, contains <span class="op">=</span> <span class="st">"Person"</span><span class="op">)</span>

<span class="va">person1</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"Person"</span>, name<span class="op">=</span><span class="st">"Peter"</span>, age<span class="op">=</span><span class="fl">34</span><span class="op">)</span>
<span class="va">boss1</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"Employee"</span>, boss<span class="op">=</span><span class="va">person1</span>, name<span class="op">=</span><span class="st">"Lisa"</span>, age<span class="op">=</span><span class="fl">36</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">person1</span></code></pre></div>
<pre><code><span class="co">## An object of class "Person"</span>
<span class="co">## Slot "name":</span>
<span class="co">## [1] "Peter"</span>
<span class="co">## </span>
<span class="co">## Slot "age":</span>
<span class="co">## [1] 34</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">boss1</span></code></pre></div>
<pre><code><span class="co">## An object of class "Employee"</span>
<span class="co">## Slot "boss":</span>
<span class="co">## An object of class "Person"</span>
<span class="co">## Slot "name":</span>
<span class="co">## [1] "Peter"</span>
<span class="co">## </span>
<span class="co">## Slot "age":</span>
<span class="co">## [1] 34</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## Slot "name":</span>
<span class="co">## [1] "Lisa"</span>
<span class="co">## </span>
<span class="co">## Slot "age":</span>
<span class="co">## [1] 36</span></code></pre>
<p>Example class: SummarizedExperiment</p>
<p>Representation:</p>
<p><code>RangedSummarizedExperiment</code> is an extension of <code>SummarizedExperiment</code>. It uses <code>GRanges</code> to define the genomic regions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required package: MatrixGenerics</span></code></pre>
<pre><code><span class="co">## Loading required package: matrixStats</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'MatrixGenerics'</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:matrixStats':</span>
<span class="co">## </span>
<span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="co">##     rowWeightedSds, rowWeightedVars</span></code></pre>
<pre><code><span class="co">## Loading required package: GenomicRanges</span></code></pre>
<pre><code><span class="co">## Loading required package: stats4</span></code></pre>
<pre><code><span class="co">## Loading required package: BiocGenerics</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'BiocGenerics'</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:stats':</span>
<span class="co">## </span>
<span class="co">##     IQR, mad, sd, var, xtabs</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">##     union, unique, unsplit, which.max, which.min</span></code></pre>
<pre><code><span class="co">## Loading required package: S4Vectors</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'S4Vectors'</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     expand.grid, I, unname</span></code></pre>
<pre><code><span class="co">## Loading required package: IRanges</span></code></pre>
<pre><code><span class="co">## Loading required package: GenomeInfoDb</span></code></pre>
<pre><code><span class="co">## Loading required package: Biobase</span></code></pre>
<pre><code><span class="co">## Welcome to Bioconductor</span>
<span class="co">## </span>
<span class="co">##     Vignettes contain introductory material; view with</span>
<span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'Biobase'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:MatrixGenerics':</span>
<span class="co">## </span>
<span class="co">##     rowMedians</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:matrixStats':</span>
<span class="co">## </span>
<span class="co">##     anyMissing, rowMedians</span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#data("airway_small",  package="BasicR")</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/airway_small.rda"</span><span class="op">)</span>
<span class="va">se</span> <span class="op">&lt;-</span> <span class="va">airway_small</span></code></pre></div>
<div class="section level2">
<h2 id="accessors">Accessors<a class="anchor" aria-label="anchor" href="#accessors"></a>
</h2>
<p>To know all this, look for the help and the vignette.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DataFrame with 8 rows and 9 columns</span>
<span class="co">##            SampleName     cell      dex    albut        Run avgLength</span>
<span class="co">##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;</span>
<span class="co">## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126</span>
<span class="co">## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126</span>
<span class="co">## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126</span>
<span class="co">## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87</span>
<span class="co">## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120</span>
<span class="co">## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126</span>
<span class="co">## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101</span>
<span class="co">## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98</span>
<span class="co">##            Experiment    Sample    BioSample</span>
<span class="co">##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span>
<span class="co">## SRR1039508  SRX384345 SRS508568 SAMN02422669</span>
<span class="co">## SRR1039509  SRX384346 SRS508567 SAMN02422675</span>
<span class="co">## SRR1039512  SRX384349 SRS508571 SAMN02422678</span>
<span class="co">## SRR1039513  SRX384350 SRS508572 SAMN02422670</span>
<span class="co">## SRR1039516  SRX384353 SRS508575 SAMN02422682</span>
<span class="co">## SRR1039517  SRX384354 SRS508576 SAMN02422673</span>
<span class="co">## SRR1039520  SRX384357 SRS508579 SAMN02422683</span>
<span class="co">## SRR1039521  SRX384358 SRS508580 SAMN02422677</span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DataFrame with 4001 rows and 0 columns</span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#assay(se)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRangesList object of length 4001:</span>
<span class="co">## $ENSG00000000003</span>
<span class="co">## GRanges object with 17 ranges and 2 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |   exon_id       exon_name</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;</span>
<span class="co">##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322</span>
<span class="co">##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868</span>
<span class="co">##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072</span>
<span class="co">##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132</span>
<span class="co">##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016</span>
<span class="co">##    ...      ...               ...    ... .       ...             ...</span>
<span class="co">##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331</span>
<span class="co">##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883</span>
<span class="co">##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382</span>
<span class="co">##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395</span>
<span class="co">##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 722 sequences (1 circular) from an unspecified genome</span>
<span class="co">## </span>
<span class="co">## ...</span>
<span class="co">## &lt;4000 more elements&gt;</span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#assays(se)[[1]]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516</span>
<span class="co">## ENSG00000000003        679        448        873        408       1138</span>
<span class="co">## ENSG00000000005          0          0          0          0          0</span>
<span class="co">## ENSG00000000419        467        515        621        365        587</span>
<span class="co">## ENSG00000000457        260        211        263        164        245</span>
<span class="co">## ENSG00000000460         60         55         40         35         78</span>
<span class="co">## ENSG00000000938          0          0          2          0          1</span>
<span class="co">##                 SRR1039517 SRR1039520 SRR1039521</span>
<span class="co">## ENSG00000000003       1047        770        572</span>
<span class="co">## ENSG00000000005          0          0          0</span>
<span class="co">## ENSG00000000419        799        417        508</span>
<span class="co">## ENSG00000000457        331        233        229</span>
<span class="co">## ENSG00000000460         63         76         60</span>
<span class="co">## ENSG00000000938          0          0          0</span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">airway_small</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [[1]]</span>
<span class="co">## Experiment data</span>
<span class="co">##   Experimenter name: Himes BE </span>
<span class="co">##   Laboratory: NA </span>
<span class="co">##   Contact information:  </span>
<span class="co">##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. </span>
<span class="co">##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 </span>
<span class="co">##   PMIDs: 24926665 </span>
<span class="co">## </span>
<span class="co">##   Abstract: A 226 word abstract is available. Use 'abstract' method.</span></code></pre>
<p>You can access each data “natively”, and even change it, but you shouldn’t do that. Why? It is easy to screw it up…</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span></code></pre></div>
<pre><code><span class="co">## GRangesList object of length 4001:</span>
<span class="co">## $ENSG00000000003</span>
<span class="co">## GRanges object with 17 ranges and 2 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |   exon_id       exon_name</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;</span>
<span class="co">##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322</span>
<span class="co">##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868</span>
<span class="co">##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072</span>
<span class="co">##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132</span>
<span class="co">##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016</span>
<span class="co">##    ...      ...               ...    ... .       ...             ...</span>
<span class="co">##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331</span>
<span class="co">##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883</span>
<span class="co">##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382</span>
<span class="co">##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395</span>
<span class="co">##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 722 sequences (1 circular) from an unspecified genome</span>
<span class="co">## </span>
<span class="co">## ...</span>
<span class="co">## &lt;4000 more elements&gt;</span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span><span class="op">@</span><span class="va">metadata</span></code></pre></div>
<pre><code><span class="co">## [[1]]</span>
<span class="co">## Experiment data</span>
<span class="co">##   Experimenter name: Himes BE </span>
<span class="co">##   Laboratory: NA </span>
<span class="co">##   Contact information:  </span>
<span class="co">##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. </span>
<span class="co">##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 </span>
<span class="co">##   PMIDs: 24926665 </span>
<span class="co">## </span>
<span class="co">##   Abstract: A 226 word abstract is available. Use 'abstract' method.</span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">###don't do this:</span>
<span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3000</span><span class="op">]</span>
<span class="va">se</span></code></pre></div>
<pre><code><span class="co">## class: RangedSummarizedExperiment </span>
<span class="co">## dim: 4001 8 </span>
<span class="co">## metadata(1): ''</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(3000): ENSG00000000003 ENSG00000000005 ... ENSG00000162951</span>
<span class="co">##   ENSG00000162959</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span>
<span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span></code></pre></div>
<pre><code><span class="co">## GRangesList object of length 3000:</span>
<span class="co">## $ENSG00000000003</span>
<span class="co">## GRanges object with 17 ranges and 2 metadata columns:</span>
<span class="co">##        seqnames            ranges strand |   exon_id       exon_name</span>
<span class="co">##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;</span>
<span class="co">##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322</span>
<span class="co">##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868</span>
<span class="co">##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072</span>
<span class="co">##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132</span>
<span class="co">##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016</span>
<span class="co">##    ...      ...               ...    ... .       ...             ...</span>
<span class="co">##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331</span>
<span class="co">##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883</span>
<span class="co">##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382</span>
<span class="co">##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395</span>
<span class="co">##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 722 sequences (1 circular) from an unspecified genome</span>
<span class="co">## </span>
<span class="co">## ...</span>
<span class="co">## &lt;2999 more elements&gt;</span></code></pre>
</div>
<div class="section level2">
<h2 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h2>
<div class="section level3">
<h3 id="subsetting">subsetting<a class="anchor" aria-label="anchor" href="#subsetting"></a>
</h3>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subsetting is very easy, just like with a data.frame</span>

<span class="va">se</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2000</span>,<span class="op">]</span>

<span class="va">se</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>

<span class="va">se</span></code></pre></div>
<pre><code><span class="co">## class: RangedSummarizedExperiment </span>
<span class="co">## dim: 2000 3 </span>
<span class="co">## metadata(1): ''</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(2000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758</span>
<span class="co">##   ENSG00000092820</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(3): SRR1039508 SRR1039509 SRR1039512</span>
<span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></code></pre>
</div>
<div class="section level3">
<h3 id="combining">Combining<a class="anchor" aria-label="anchor" href="#combining"></a>
</h3>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rowbinded</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">se</span><span class="op">)</span>
<span class="va">rowbinded</span></code></pre></div>
<pre><code><span class="co">## class: RangedSummarizedExperiment </span>
<span class="co">## dim: 4000 3 </span>
<span class="co">## metadata(2): '' ''</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(4000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758</span>
<span class="co">##   ENSG00000092820</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(3): SRR1039508 SRR1039509 SRR1039512</span>
<span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">colbinded</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">se</span><span class="op">)</span>
<span class="va">colbinded</span></code></pre></div>
<pre><code><span class="co">## class: RangedSummarizedExperiment </span>
<span class="co">## dim: 2000 6 </span>
<span class="co">## metadata(2): '' ''</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(2000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758</span>
<span class="co">##   ENSG00000092820</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(6): SRR1039508 SRR1039509 ... SRR1039509 SRR1039512</span>
<span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="st">"X"</span>, ranges<span class="op">=</span><span class="fl">99800000</span><span class="op">:</span><span class="fl">99850000</span><span class="op">)</span>
<span class="va">se_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">roi</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="granges-and-grangeslist">Granges and GRangesList<a class="anchor" aria-label="anchor" href="#granges-and-grangeslist"></a>
</h2>
<p><code>GRanges</code> objects store information about Genomic ranges with chromosome, position and additional information. Offers a wide range of functionality. <code>GRangesList</code> is a “grouped” version of <code>GRnages</code>, it is similar to a normal list, but with additional functionality. Both of them are able to handle strand information as well.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span>

<span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>
    seqnames <span class="op">=</span> <span class="st">"chr2"</span>,
    ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fl">103</span>, <span class="fl">106</span><span class="op">)</span>,
    strand <span class="op">=</span> <span class="st">"+"</span>,
    score <span class="op">=</span> <span class="fl">5L</span>, GC <span class="op">=</span> <span class="fl">0.45</span><span class="op">)</span>
<span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>
    seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr3"</span><span class="op">)</span>,
    ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">107</span>, <span class="fl">113</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,
    strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span><span class="op">)</span>,
    score <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">4</span>, GC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">gr3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr2:102-107"</span><span class="op">)</span>

<span class="va">gr4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>, 
                                           start<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">104</span>, <span class="fl">104</span><span class="op">)</span>, end<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">108</span>, <span class="fl">105</span><span class="op">)</span>,
                                           name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene1"</span>, <span class="st">"gene2"</span><span class="op">)</span><span class="op">)</span>, keep.extra.columns <span class="op">=</span> <span class="cn">T</span>, ignore.strand <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a></span><span class="op">(</span><span class="st">"txA"</span> <span class="op">=</span> <span class="va">gr1</span>, <span class="st">"txB"</span> <span class="op">=</span> <span class="va">gr2</span><span class="op">)</span>
<span class="va">grl</span></code></pre></div>
<pre><code><span class="co">## GRangesList object of length 2:</span>
<span class="co">## $txA</span>
<span class="co">## GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span>
<span class="co">## </span>
<span class="co">## $txB</span>
<span class="co">## GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">##   [2]     chr3   113-115      - |         4       0.5</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#seqinfo(gr1) &lt;- Seqinfo(genome="hg38")</span>
<span class="co">#can have names for each row</span></code></pre></div>
<div class="section level3">
<h3 id="accessors-1">Accessors<a class="anchor" aria-label="anchor" href="#accessors-1"></a>
</h3>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## factor-Rle of length 1 with 1 run</span>
<span class="co">##   Lengths:    1</span>
<span class="co">##   Values : chr2</span>
<span class="co">## Levels(1): chr2</span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "chr2"</span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/range-squeezers.html" class="external-link">ranges</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## IRanges object with 1 range and 0 metadata columns:</span>
<span class="co">##           start       end     width</span>
<span class="co">##       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">##   [1]       103       106         4</span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 103</span></code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 106</span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## factor-Rle of length 1 with 1 run</span>
<span class="co">##   Lengths: 1</span>
<span class="co">##   Values : +</span>
<span class="co">## Levels(3): + - *</span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 4</span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 1</span></code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">gr4</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DataFrame with 2 rows and 1 column</span>
<span class="co">##          name</span>
<span class="co">##   &lt;character&gt;</span>
<span class="co">## 1       gene1</span>
<span class="co">## 2       gene2</span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">gr4</span><span class="op">)</span><span class="op">$</span><span class="va">name</span></code></pre></div>
<pre><code><span class="co">## [1] "gene1" "gene2"</span></code></pre>
</div>
<div class="section level3">
<h3 id="subsetting-1">Subsetting<a class="anchor" aria-label="anchor" href="#subsetting-1"></a>
</h3>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#data.frame-like</span>

<span class="va">gr1</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gr4</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"name"</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        name</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">##   [1]     chr1   104-108      * |       gene1</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset function</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">strand</span> <span class="op">==</span> <span class="st">"+"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre>
</div>
<div class="section level3">
<h3 id="splitting-and-combining">Splitting and combining<a class="anchor" aria-label="anchor" href="#splitting-and-combining"></a>
</h3>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">gr4</span>, <span class="op">~</span> <span class="va">name</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRangesList object of length 2:</span>
<span class="co">## $gene1</span>
<span class="co">## GRanges object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        name</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">##   [1]     chr1   104-108      * |       gene1</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">## </span>
<span class="co">## $gene2</span>
<span class="co">## GRanges object with 1 range and 1 metadata column:</span>
<span class="co">##       seqnames    ranges strand |        name</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">##   [1]     chr2   104-105      * |       gene2</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">sp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">sp</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">##   [2]     chr3   113-115      - |         4       0.5</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/stack.html" class="external-link">stack</a></span><span class="op">(</span><span class="va">sp</span>, index.var<span class="op">=</span><span class="st">"name"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 2 ranges and 3 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |  name     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;Rle&gt; &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr1   107-109      + |     1         3       0.3</span>
<span class="co">##   [2]     chr3   113-115      - |     2         4       0.5</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre>
</div>
<div class="section level3">
<h3 id="aggregating">Aggregating<a class="anchor" aria-label="anchor" href="#aggregating"></a>
</h3>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/aggregate-methods.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="va">score</span> <span class="op">~</span> <span class="va">strand</span>, <span class="va">mean</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DataFrame with 2 rows and 2 columns</span>
<span class="co">##     strand     score</span>
<span class="co">##   &lt;factor&gt; &lt;numeric&gt;</span>
<span class="co">## 1        +         3</span>
<span class="co">## 2        -         4</span></code></pre>
</div>
<div class="section level3">
<h3 id="interval-operations">Interval operations<a class="anchor" aria-label="anchor" href="#interval-operations"></a>
</h3>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">flank</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="fl">10</span>, both<span class="op">=</span><span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr2    93-112      + |         5      0.45</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">shift</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr1   112-114      + |         3       0.3</span>
<span class="co">##   [2]     chr3   118-120      - |         4       0.5</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr2   103-122      + |         5      0.45</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr3</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">##       seqnames    ranges strand |     score        GC</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">##   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">##   [2]     chr2   102-107      * |      &lt;NA&gt;        NA</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr3</span><span class="op">)</span><span class="op">)</span> <span class="co"># combines the overlapping ranges</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">##       seqnames    ranges strand</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">##   [1]     chr2   103-106      +</span>
<span class="co">##   [2]     chr2   102-107      *</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr3</span><span class="op">)</span>, ignore.strand<span class="op">=</span><span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 1 range and 0 metadata columns:</span>
<span class="co">##       seqnames    ranges strand</span>
<span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">##   [1]     chr2   102-107      *</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre>
</div>
<div class="section level3">
<h3 id="comapring-2-granges">Comapring 2 GRanges<a class="anchor" aria-label="anchor" href="#comapring-2-granges"></a>
</h3>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#union(gr2, gr4)</span>

<span class="co">#intersect(gr2, gr4)</span>
<span class="co">#setdiff(gr2, gr4)</span>

<span class="co">#hits &lt;- findOverlaps(gr2, gr4, ignore.strand=TRUE)</span>
<span class="co">#countOverlaps(gr2, gr4, ignore.strand=TRUE)</span>
<span class="co">#subsetByOverlaps(gr2, gr4)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="deseq2">DeSeq2<a class="anchor" aria-label="anchor" href="#deseq2"></a>
</h2>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'dplyr'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:Biobase':</span>
<span class="co">## </span>
<span class="co">##     combine</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:GenomicRanges':</span>
<span class="co">## </span>
<span class="co">##     intersect, setdiff, union</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:GenomeInfoDb':</span>
<span class="co">## </span>
<span class="co">##     intersect</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:IRanges':</span>
<span class="co">## </span>
<span class="co">##     collapse, desc, intersect, setdiff, slice, union</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:S4Vectors':</span>
<span class="co">## </span>
<span class="co">##     first, intersect, rename, setdiff, setequal, union</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:BiocGenerics':</span>
<span class="co">## </span>
<span class="co">##     combine, intersect, setdiff, union</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:matrixStats':</span>
<span class="co">## </span>
<span class="co">##     count</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:stats':</span>
<span class="co">## </span>
<span class="co">##     filter, lag</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     intersect, setdiff, setequal, union</span></code></pre>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">count_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"../inst/extdata/GSE139563_Processed_data_Paracrine_Senescence.txt.gz"</span>, skip<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">count_mat</span> <span class="op">&lt;-</span> <span class="va">count_mat</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"count"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"../inst/extdata//GSE139563_Processed_data_Paracrine_Senescence.txt.gz"</span>, header<span class="op">=</span><span class="cn">F</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">$</span><span class="va">V1</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>
<span class="va">names</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">" "</span>, <span class="st">"_"</span>, <span class="va">meta</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">names</span><span class="op">[</span><span class="va">names</span><span class="op">!=</span><span class="st">""</span><span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"O"</span>, <span class="st">"0"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span><span class="op">)</span>

<span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span>, tp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(D[[:digit:]]+)_(rep[12])"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span><span class="op">)</span>, 
                      rep<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(D[[:digit:]]+)_(rep[12])"</span>, <span class="st">"\\2"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_mat</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> |&gt; 
            <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>tp_cat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">tp</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"D0"</span>, <span class="st">"D4"</span><span class="op">)</span>, <span class="st">"early"</span>, <span class="st">"late"</span><span class="op">)</span><span class="op">)</span>

<span class="va">anno_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  GeneID <span class="op">=</span> <span class="va">meta</span><span class="op">$</span><span class="va">V1</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>,
  gene_name <span class="op">=</span> <span class="va">meta</span><span class="op">$</span><span class="va">V2</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>,
  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>,
  row.names <span class="op">=</span> <span class="va">meta</span><span class="op">$</span><span class="va">V1</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>
<span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span>countData <span class="op">=</span> <span class="va">count_mat</span>,
                              colData <span class="op">=</span> <span class="va">coldata</span>,
                              design <span class="op">=</span> <span class="op">~</span> <span class="va">tp_cat</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span>
<span class="co">## design formula are characters, converting to factors</span></code></pre>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds</span></code></pre></div>
<pre><code><span class="co">## class: DESeqDataSet </span>
<span class="co">## dim: 27696 8 </span>
<span class="co">## metadata(1): version</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(27696): ENSG00000238009.6 ENSG00000268903.1 ...</span>
<span class="co">##   ENSG00000210195.2 ENSG00000210196.2</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(8): D0_rep1 D0_rep2 ... D10_rep1 D10_rep2</span>
<span class="co">## colData names(3): tp rep tp_cat</span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## estimating size factors</span></code></pre>
<pre><code><span class="co">## estimating dispersions</span></code></pre>
<pre><code><span class="co">## gene-wise dispersion estimates</span></code></pre>
<pre><code><span class="co">## mean-dispersion relationship</span></code></pre>
<pre><code><span class="co">## final dispersion estimates</span></code></pre>
<pre><code><span class="co">## fitting model and testing</span></code></pre>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> 
<span class="co">#res &lt;- res[which(res$padj&lt;0.05),]</span>

<span class="va">ge_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/varianceStabilizingTransformation.html" class="external-link">getVarianceStabilizedData</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hasaru-k/GlimmaV2" class="external-link">Glimma</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/Glimma/man/glMDSPlot.html" class="external-link">glMDSPlot</a></span><span class="op">(</span><span class="va">ge_mat</span>,  groups<span class="op">=</span><span class="va">coldata</span><span class="op">$</span><span class="va">tp</span>, labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span>, launch<span class="op">=</span><span class="cn">T</span>, folder <span class="op">=</span> <span class="st">"/glimma-plots"</span>, html <span class="op">=</span> <span class="st">"MDS-Plot_full"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R">    <span class="fu"><a href="https://rdrr.io/pkg/Glimma/man/glMDPlot.html" class="external-link">glMDPlot</a></span><span class="op">(</span><span class="va">res</span>, counts<span class="op">=</span><span class="va">ge_mat</span>, groups <span class="op">=</span> <span class="va">dds</span><span class="op">@</span><span class="va">colData</span><span class="op">[</span>,<span class="st">"tp"</span><span class="op">]</span>, anno<span class="op">=</span><span class="va">anno_df</span>, main <span class="op">=</span> <span class="st">"Results"</span>, status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span>, launch <span class="op">=</span> <span class="cn">T</span>, folder <span class="op">=</span> <span class="st">"/glimma-plots"</span> , transform <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">ge_mat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span><span class="op">]</span>,<span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="S4_classes_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">ge_mat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span><span class="op">]</span>,<span class="op">]</span>, scale <span class="op">=</span> <span class="st">"row"</span>, show_rownames <span class="op">=</span> <span class="cn">F</span>, annotation_col <span class="op">=</span> <span class="va">coldata</span><span class="op">)</span></code></pre></div>
<p><img src="S4_classes_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">clusterProfiler</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span></code></pre>
<pre><code><span class="co">## Registered S3 method overwritten by 'ggtree':</span>
<span class="co">##   method      from </span>
<span class="co">##   identify.gg ggfun</span></code></pre>
<pre><code><span class="co">## clusterProfiler v4.2.2  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span>
<span class="co">## </span>
<span class="co">## If you use clusterProfiler in published research, please cite:</span>
<span class="co">## T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'clusterProfiler'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:IRanges':</span>
<span class="co">## </span>
<span class="co">##     slice</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:S4Vectors':</span>
<span class="co">## </span>
<span class="co">##     rename</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:stats':</span>
<span class="co">## </span>
<span class="co">##     filter</span></code></pre>
<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loading required package: AnnotationDbi</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'AnnotationDbi'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:clusterProfiler':</span>
<span class="co">## </span>
<span class="co">##     select</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:dplyr':</span>
<span class="co">## </span>
<span class="co">##     select</span></code></pre>
<pre><code><span class="co">## </span></code></pre>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene          <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(ENSG[[:digit:]]*)(\\.[[:digit:]]+)"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span><span class="op">]</span>,
                universe      <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(ENSG[[:digit:]]*)(\\.[[:digit:]]+)"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span>,
                OrgDb         <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,
                keyType <span class="op">=</span> <span class="st">"ENSEMBL"</span>,
                ont           <span class="op">=</span> <span class="st">"MF"</span>,
                pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,
                pvalueCutoff  <span class="op">=</span> <span class="fl">0.01</span>,
                qvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span>,
        readable      <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Reka Toth, Rajbir Nath Batra.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
